### ISSUE TITLE
[Feature Request] train-text-from-scratch.cpp : GPU support and fine tuning

### ISSUE THREAD

**Issue by YerongLi** [ggerganov/llama.cpp#2318]
Can we add fine-tuning function for train-text-from-scratch.cpp ? {prompted text : label}
I assume train-text-from-scratch.cpp does not support GPU pretraining, can we add support to this?
